clog/loggers.py

```diff
@@ -295,27 +304,57 @@ def log_line(self, stream, line):
             self.metrics.monk_exception()
 
     def _log_line_no_size_limit(self, stream, line):
-        if time.time() < self.last_timeout + self.timeout_backoff_s:
+        now = time.time()
+        if now - self.last_disconnect < self.timeout_backoff_s:
+            if self.buffering:
```

Comment:
I'm not sure I understand why we need this extra variable. Here we just need to add the line to the buffer if buffering is enabled (which is checked in `_add_to_buffer`). In the `elif` part we could just use the size of the deque, or even just call the `_flush_buffer` method.